tests/integration/test_deploy_aws.py

```diff
@@ -0,0 +1,26 @@
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+CHARM_NAME = METADATA["name"]
+
+
+class TestDeployAWS:
```

Comment:
Haven't looked / just a fly-by comment:

Would you be able to keep only one tox env using a decorator such as
`@skipIf(os.environ.get("AWS_SOMETHING"), "Skipping because...")`

For example, we [can detect](https://github.com/canonical/traefik-k8s-operator/blob/a287fa6a41077c8c3b7d3eb244f055c2a354bd2a/tests/integration/conftest.py#L42) if we're running in a GH runner.

Reply:
I think @sed-i is on to something with this.  Feels like something that could be environment-aware

Although whether it is worth it, that's another question